crates/wasmtime/src/linker.rs

```diff
@@ -401,63 +522,57 @@ impl Linker {
+    pub fn module(
+        &mut self,
+        mut store: impl AsContextMut<Data = T>,
+        module_name: &str,
+        module: &Module,
+    ) -> Result<&mut Self> {
         match ModuleKind::categorize(module)? {
-            ModuleKind::Command => self.command(module_name, module),
+            ModuleKind::Command => self.command(store, module_name, module),
             ModuleKind::Reactor => {
-                let instance = self.instantiate(&module)?;
+                let instance = self.instantiate(&mut store, &module)?;
 
-                if let Some(export) = instance.get_export("_initialize") {
+                if let Some(export) = instance.get_export(&mut store, "_initialize") {
                     if let Extern::Func(func) = export {
-                        func.typed::<(), ()>()
-                            .and_then(|f| f.call(()).map_err(Into::into))
+                        func.typed::<(), (), _>(&store)
+                            .and_then(|f| f.call(&mut store, ()).map_err(Into::into))
                             .context("calling the Reactor initialization function")?;
                     }
                 }
 
-                self.instance(module_name, &instance)
+                self.instance(store, module_name, instance)
             }
         }
     }
 
-    fn command(&mut self, module_name: &str, module: &Module) -> Result<&mut Self> {
+    fn command(
+        &mut self,
+        mut store: impl AsContextMut<Data = T>,
+        module_name: &str,
+        module: &Module,
+    ) -> Result<&mut Self> {
         for export in module.exports() {
             if let Some(func_ty) = export.ty().func() {
-                let imports = self
-                    .compute_imports(module)?
-                    .into_iter()
-                    .map(|e| e.wasmtime_export())
-                    .collect::<Vec<_>>();
+                let imports = self.compute_imports(&mut store, module)?;
                 let module = module.clone();
                 let export_name = export.name().to_owned();
                 let func = Func::new(
-                    &self.store,
+                    &mut store,
                     func_ty.clone(),
-                    move |caller, params, results| {
-                        let store = caller.store();
-
-                        // Note that the unsafety here is due to the validity of
-                        // `i` and the validity of `i` within `store`. For our
-                        // case though these items all come from `imports` above
-                        // so they're all valid. They're also all kept alive by
-                        // the store itself used here so this should be safe.
-                        let imports = imports
-                            .iter()
-                            .map(|i| unsafe { Extern::from_wasmtime_export(&i, &store) })
-                            .collect::<Vec<_>>();
```

Comment:
This was a very satisfying thing to remove. This was originally added to prevent cycles, but now there are no cycles, so we could just delete everything related to this!